### ISSUE TITLE
A better way to specify notes?

### ISSUE THREAD

**Issue by ndim** [spkr-beep/beep#19]
Traditionally, `beep` has been using a combination of the `-f`, `-l`, `-D`, `-d`, and `-n` parameters to specify notes and sequences of notes.

I am certain there are better ways to write notes in a way which is both less verbose, and easier to read and write at the same time. I am also certain that over the last 20 to 40 years, there must have been utilities using notations which at least give good ideas.

This issue is to discuss and document better ways to write notes than with , and to discuss integrating a better note notation into `beep`.

**Comment by dawidi** [comment]
Here's what I did with shell script wrappers around `beep`:
[sound](https://github.com/spkr-beep/beep/files/6212432/sound.txt)
This is just a slightly more elegant way to specify `frequency,duration` pairs - named after the `SOUND` command in BASIC.
[spkplay](https://github.com/spkr-beep/beep/files/6212431/spkplay.txt)
This interprets a music-oriented notation similar to the `PLAY` command in BASIC, setting the tempo, note lengths, octaves, staccato/legato playing styles, sharp/flat notes etc.
Like so:
`spkplay t114 mn l16 o4 c d f d l8 a. a. l4 g ms l16 c d f d l8 g. g. f mn l16 e l8 d l16 c d f d l4 f l8 g e. l16 d l4 c l8 c l4 g l2 f`

**Comment by ndim** [comment]
Also, a quick remark on the command line argument checks:

```
if [[ "$@" == "" ]]; then
```

The `"$@"` expands to multiple arguments if the script is called with multiple arguments like `script.bash foo bar bla`, which results in the test expression being evaluated to be `[[ "foo" "bar" "bla" == "" ]]` which does not appear to be what you intended here.

Are you sure you did not want to check `"$*" == ""` here instead, or maybe even `"$#" -lt 1`?

**Comment by dawidi** [comment]
Indeed, `"$#" -lt 1` would be better, that's what I meant to do. Thanks!

**Comment by ndim** [comment]
The BSDs appears to have a kernel implementation of an IBM BASIC way of specifying notes, which the `speaker(4)` man page describes pretty well:

https://www.freebsd.org/cgi/man.cgi?query=speaker&sektion=4&manpath=FreeBSD+13.0-RELEASE+and+Ports
https://man.netbsd.org/speaker.4
